lib/Utils/PlatformMac.cpp

```diff
@@ -0,0 +1,214 @@
+                 << " not found, this would " << Verbose << "\n";
+
+  return false;
+}
+
+static std::string ReadSingleLine(const char* Cmd) {
+  if (FILE* PF = ::popen(Cmd, "r")) {
+    char Buf[1024];
+    char* BufPtr = ::fgets(Buf, sizeof(Buf), PF);
+    ::pclose(PF);
+    if (BufPtr && Buf[0]) {
+      const llvm::StringRef Result(Buf);
+      assert(Result[Result.size()-1] == '\n' && "Single line too large");
+      return Result.trim().str();
+    }
+  }
+  return "";
+}
+
+} // anonymous namespace
+
+bool GetISysRoot(std::string& sysRoot, bool Verbose) {
+  using namespace llvm::sys;
+
+  // Some versions of OS X and Server have headers installed
+  if (fs::is_regular_file("/usr/include/stdlib.h"))
+    return false;
+
+  std::string SDKs("/Applications/Xcode.app/Contents/Developer");
+
+  // Is XCode installed where it usually is?
+  if (!fs::is_directory(SDKs)) {
+    // Nope, use xcode-select -p to get the path
+    SDKs = ReadSingleLine("xcode-select -p");
+    if (SDKs.empty())
+      return false;  // Nothing more we can do
+  }
+
+  SDKs.append("/Platforms/MacOSX.platform/Developer/SDKs/");
+  if (!fs::is_directory(SDKs))
+    return false;
+
+
+  // Try to get the SDK for whatever version of OS X is currently running
+  // Seems to make more sense to get the currently running SDK so headers
+  // and any loaded libraries will match.
+
+  int32_t majorVers = -1, minorVers = -1;
+#ifndef CLING_SWVERS_PARSE_ONLY
+ #pragma clang diagnostic push
+ #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+  if (void *core = ::dlopen(
+          "/System/Library/Frameworks/CoreServices.framework/CoreServices",
+          RTLD_LAZY)) {
+    typedef ::OSErr (*GestaltProc)(::OSType, ::SInt32 *);
+    if (GestaltProc Gestalt = (GestaltProc)::dlsym(core, "Gestalt")) {
+      if (Gestalt(gestaltSystemVersionMajor, &majorVers) == ::noErr) {
+        if (Gestalt(gestaltSystemVersionMinor, &minorVers) != ::noErr)
+          minorVers = -1;
+      } else
+        majorVers = -1;
+    }
+    ::dlclose(core);
+  }
+ #pragma clang diagnostic pop
+#endif
+
+  if (majorVers == -1 || minorVers == -1) {
+    const std::string SWVers = ReadSingleLine("sw_vers | grep ProductVersion"
+                                              " | awk '{print $2}'");
+    if (!SWVers.empty()) {
+      if (::sscanf(SWVers.c_str(), "%d.%d", &majorVers, &minorVers) != 2) {
+        majorVers = -1;
+        minorVers = -1;
+      }
+    }
+  }
+
+  if (majorVers != -1 && minorVers != -1) {
+    if (getISysRootVersion(SDKs, majorVers, minorVers, sysRoot,
+            Verbose ? "match the version of OS X running"
+                    : nullptr)) {
+      return true;
+    }
+  }
+
+
+#define GET_ISYSROOT_VER(maj, min) \
+  if (getISysRootVersion(SDKs, maj, min, sysRoot, Verbose ? \
+                 "match what cling was compiled with" : nullptr)) \
+    return true;
+
+  // Try to get the SDK for whatever cling was compiled with
+  #if defined(MAC_OS_X_VERSION_10_11)
```

Comment:
But we can just `#undef` those before including the apple version magic. Would that work? Can you do it?